src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -116,14 +119,14 @@ export class DeviceDetails extends Component {
               <Accordion title={t('devices.details.tags.title')} description={t('devices.details.tags.description')}>
                 <Grid>
                   {
-                    (device.tags && device.tags.length) &&
+                    (tags.length > 0) &&
```

Comment:
Didn't notice this a second ago but wrap the grid itself with the `(tags.length > 0)` check and if nothing is there display some text saying there are no tags. Same the for content below here.